frontend/src/app/modules/work_packages/routing/wp-full-view/wp-full-view.component.ts

```diff
@@ -50,18 +52,28 @@ export class WorkPackagesFullViewComponent extends WorkPackageSingleViewBase imp
   public displayWatchButton:boolean;
   public watchers:any;
 
+  // additional tabs
+  private registeredAdditionalTabs:Tab[];
+
   stateName$ = of('work-packages.new');
 
   constructor(public injector:Injector,
               public wpTableSelection:WorkPackageViewSelectionService,
-              readonly $state:StateService) {
+              readonly $state:StateService,
+              readonly HookService:HookService) {
     super(injector, $state.params['workPackageId']);
+
+    this.registeredAdditionalTabs = this.HookService.call('workPackageAdditionalTabs');
```

Comment:
not sure if it's worth to cache the result of this hook call 🤔  we could inline this to `additionalTabs()` below if not.